packages/@react-aria/dnd/src/useDroppableCollection.ts

```diff
@@ -59,8 +59,8 @@ export function useDroppableCollection(props: DroppableCollectionOptions, state:
       shouldAcceptItemDrop
     } = localState.props;
 
-    let {draggingCollectionRef, draggingKeys} = globalDndState;
-    let isInternalDrop = draggingCollectionRef?.current != null && draggingCollectionRef?.current === ref?.current;
+    let {draggingKeys} = globalDndState;
+    let isInternal = isInternalDropOperation(ref);
```

Comment:
A bunch of these `isInternal` checks use `ref` instead of relying on the global state's `dropCollectionRef` because I felt it was safer this way (global state is spooky in general). Since these are all within `useDroppableCollection`, the ref should be equal to dropCollectionRef for the most part (some exceptions exist where we want to calculate isInternal before we even set dropCollectionRef)